taxonomy-editor-frontend/src/pages/editentry/AccumulateAllComponents.jsx

```diff
@@ -14,46 +15,104 @@ import ListAllEntryRelationships from "./ListAllEntryRelationships";
  * If node is "header/footer": Comments are rendered  
 */ 
 const AccumulateAllComponents = ({ id }) => {
+
+    // Finding URL to send requests
     const url = createURL(id);
     const isEntry = getIdType(id) === 'entry';
-    const [nodeObject, setNodeObject] = useState(null); // Storing node information
+
+    const [nodeObject, setNodeObject] = useState(null); // Storing original node information
+    const [updatedNodeObject, setUpdatedNodeObject] = useState(null); // Storing updates to node
+    const [updateChildren, setUpdateChildren] = useState([]); // Storing updates of children in node
+    const [open, setOpen] = useState(false); // Used for Dialog component
+    const navigate = useNavigate(); // Navigation between pages
     const { data: node, isPending, isError, isSuccess, errorMessage } = useFetch(url);
 
     // Setting state of node after fetch
     useEffect(() => {
         setNodeObject(node?.[0]);
+        setUpdatedNodeObject(node?.[0]);
     }, [node])
 
-    // Check error in fetch
+    // Displaying errorMessages if any
     if (isError) {
-        return (
-            <Box>
-                <Typography sx={{ml: 4}} variant='h5'>{errorMessage}</Typography>
-            </Box>
-        )
+        return (<Typography sx={{ml: 4}} variant='h5'>{errorMessage}</Typography>)
     }
+
+    // Loading...
     if (isPending) {
-        return (
-            <Box>
-                <Typography sx={{ml: 4}} variant='h5'>Loading..</Typography>
-            </Box>
-        )
+        return (<Typography sx={{ml: 4}} variant='h5'>Loading..</Typography>)
+    }
+    
+    // Helper functions for Dialog component 
+    const handleClose = () => {setOpen(false)};
+    const handleBack = () => {navigate('/entry')};
+
+    // Function handling updation of node
+    const handleSubmit = () => {
+        const {id, ...data} = updatedNodeObject // ID not allowed in POST
+        fetch(url, {
```

Comment:
would it make sense to use `Promise.all` that way you can handle a single catch and a single success (`setOpen(true)`)